pkg/route/api/validation/validation.go

```diff
@@ -179,3 +240,71 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
 
 	return nil
 }
+
+// validatePrivateKeyForCert checks if a private key PEM is valid and
+// is valid for the certificate (if any).
+func validatePrivateKeyForCert(keyPEM, certPEM string) error {
+	if err := validatePrivateKeyPEM(keyPEM); err != nil {
+		return err
+	}
+
+	if len(certPEM) > 0 {
+		if _, err := tls.X509KeyPair([]byte(certPEM), []byte(keyPEM)); err != nil {
```

Comment:
Yes, so that's what the `tls.X509KeyPair` checks. I am currently reworking this PR to simplify it - so hopefully the next push[es] tonight/tomorrow would make this cleaner and simpler.